### PR TITLE
FIX #4955 : updated page count on enter key press

### LINKED ISSUE #4955: [Bug] Table widget doesn't update when changing the page number and hitting the return key
## Description

In table widget when you change the page number, it doesn't update until the input looses focus.
Expected behaviour: Hitting enter should update the page.

### Steps to reproduce the behaviour:

- Create a table with at least 10 pages.
- Update the page number and hit enter, it doesn't react.

Table navigation issue

### PR DESCRIPTION
**Pull request by vibhandikyash**
## Description

> updated page count on enter key press

Fixes #4955 

## Type of change

- Bug fix (non-breaking change which fixes an issue)

## How Has This Been Tested?

> Tested locally.

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

## Test coverage results :test_tube:
<details><summary>:green_circle: Total coverage has increased</summary>

    // Code coverage diff between base branch:release and head branch: FIX/4955-update-pagination-on-enter-table-widget 
Status | File | % Stmts | % Branch | % Funcs | % Lines 
 -----|-----|---------|----------|---------|------ 
 :green_circle: | total | 54.62 **(0)** | 36.7 **(0.01)** | 33.04 **(0)** | 55.23 **(0)**
 :green_circle: | app/client/src/components/designSystems/appsmith/TableComponent/TableHeader.tsx | 50 **(1.28)** | 54.17 **(2.17)** | 30 **(-3.33)** | 47.37 **(1.42)**
 :green_circle: | app/client/src/utils/autocomplete/TernServer.ts | 50.71 **(0.24)** | 38.6 **(0.88)** | 36.21 **(0)** | 55.35 **(0.27)**</details>

### CODE REVIEW

File: app/client/src/components/designSystems/appsmith/TableComponent/TableHeader.tsx [hunk 1]
```diff
@@ -52,6 +52,22 @@ function PageNumberInput(props: {
...
   }, [props.pageNo]);
+  const handleUpdatePageNo = useCallback(
+    (e) => {
+      const oldPageNo = Number(props.pageNo || 0);
+      const value = e.target.value;
+      let page = Number(value);
+      if (isNaN(value) || Number(value) < 1) {
+        page = 1;
+      }
+      if (oldPageNo < page) {
```
Comment (SatishGandham) [inline app/client/src/components/designSystems/appsmith/TableComponent/TableHeader.tsx:63]: Less than or greater than doesn't mean previous or next page.
I can go from 20->21 or 20->40.

Please check if 20->40 should trigger next page event.

File: app/client/src/components/designSystems/appsmith/TableComponent/TableHeader.tsx [hunk 2]
```diff
@@ -52,6 +52,22 @@ function PageNumberInput(props: {
   useEffect(() => {
     setPageNumber(props.pageNo || 0);
   }, [props.pageNo]);
+  const handleUpdatePageNo = useCallback(
+    (e) => {
+      const oldPageNo = Number(props.pageNo || 0);
+      const value = e.target.value;
```
Comment (SatishGandham) [inline app/client/src/components/designSystems/appsmith/TableComponent/TableHeader.tsx:58]: Can we combine these two lines.?

**Review by SatishGandham — CHANGES_REQUESTED** [review]

**Review by SatishGandham — APPROVED** [review]

**Review by somangshu — APPROVED** [review]

**Comment by vibhandikyash** [comment]
> @vibhandikyash In above deploy preview I am not able to edit the page no properly
> 
> 1. Delete button navigate to page no-1
> 2. Entering page no-5 shows page no-15
> 3. Enter button does nothing until user click somewhere on the canvas
> 
> https://www.loom.com/share/9b4480d2976d4910b31d73d96335a2b4

@vivekverma2312 According to the validations placed any pagenumber more than the max is not allowed I am not sure how does that appear for you. Are you using the deploy preview link?

**Comment by somangshu** [comment]
@vibhandikyash @vivekverma2312 here is my observation:
- The single digit page numbers can only be reached with the arrow keys and not the numbers directly, ex 1 > 5 > 8
- Pressing enter key works correctly for me
- Pagination result is not lost until the enter key is pressed or input is blurred

**Review by somangshu — APPROVED** [review]
LGTM, Deploy preview tested as well

**Review by somangshu — APPROVED** [review]

**Comment by vivekverma2312** [comment]
DP is tested and working fine.

**Pull request merged**